backend/entities/ProjectSnapshot.cs

```diff
@@ -49,6 +60,12 @@ public class ProjectSnapshot : BaseEntity
         /// </summary>
         public decimal? NetProceeds { get; set; } // TODO: Move to metadata property.
 
+        /// <summary>
+        /// get/set - The net proceed baseline integrity
+        /// </summary>
+        [NotMapped]
+        public decimal? BaselineIntegrity { get; set; } // TODO: Move to metadata property.
```

Comment:
This isn't useful as far as I can determine as a `NotMapped` property.  If you are thinking we should store the `NetProceedsVariance` rather than recalculate each time, then it would make more sense to just add it to the table with in the DB.  It may actually make a lot of sense to store the value, but we'll need to think it through.  It would mean that a prior report could fetch the `To` snapshots and would know the variance for that moment in time without having go and fetch all the `From` snapshots and recalculating.  I kinda like adding it to the DB.

Reply:
I made this a NotMapped property so that I can calculate and return the NetProceedsVariance/BaselineIntegrity at the DAL level. I decided not to store it as I was going to need to run calculations on snapshots that aren't stored in 2 different scenarios:
1) Refreshes
2) Getting temporary snapshot data when the report **From** changes.
 I could add this at the API level instead but I don't really like doing calculations at the model level. Could also do this calculation on the frontend which I think has similar +/-

Reply:
I think your reasoning is sound.  Generally it isn't good to populate entity classes with NotMapped properties, unless those properties actually do the logic (i.e. `get { return a+b; }`).
However I'm wondering if we should make it a DB property.  If we store the variance, it doesn't need to be calculated every time we load the report.  Technically it should also be the same value every time that report is requested anyways.  Your thoughts?

Reply:
We continued this conversation in the ProjectReportService update call. Closing this out as a duplicate.